docs/webhooks.rst

```diff
@@ -1,78 +1,128 @@
+*integration* on your project's **Integrations** admin page. On this page, you
+can click **Add integration** and select the integration type you'd like to add.
+In order to set up a new webhook with your provider, you'll need the *webhook
+URL*. You can get this URL by selecting the integration you just added on the
+list of integrations.
+
+The steps you need to take to manually set up a webhook vary depending on the
+provider:
 
 GitHub
----------
+~~~~~~
 
-If your project is hosted on GitHub, you can easily add a hook that will rebuild
-your docs whenever you push updates:
+* Go to the **Settings** page for your project
+* Click **Webhooks** and then **Add webhook**
+* For **Payload URL**, use the URL of the integration on Read the Docs, found on
+  the integration detail page
+* For **Content type**, both *application/json* and
+  *application/x-www-form-urlencoded* work
+* Select **Just the push event**
+* Finish by clicking **Add webhook**
 
-* Go to the "Settings" page for your project
-* Click "Integrations & Services"
-* In the "Services" section, click "Add service"
-* In the list of available services, click "ReadTheDocs"
-* Leave "Active" checked
-* Click "Add service"
-
-.. note:: The GitHub URL in your Read the Docs project must match the URL on GitHub. The URL is case-sensitive.
-
-If you ever need to manually set the webhook on GitHub,
-you can point it at ``https://readthedocs.org/github``.
+.. note:: The webhook secret is not yet respected
 
 Bitbucket
------------
+~~~~~~~~~
+
+* Go to the **Settings** page for your project
+* Click **Webhooks** and then **Add webhook**
+* For **URL**, use the URL of the integration on Read the Docs, found on the
+  integration detail page
+* Under **Triggers**, **Repository push** should be selected
+* Finish by clicking **Save**
 
-If your project is hosted on Bitbucket, you can easily add a hook that will rebuild
-your docs whenever you push updates:
+GitLab
+~~~~~~
 
-* Go to the "admin" page for your project
-* Click "Services"
-* In the available service hooks, select "Read the Docs"
-* Click "Add service"
+* Go to the **Settings** page for your project
+* Click **Integrations**
+* For **URL**, use the URL of the integration on Read the Docs, found on the
+  integration detail page
+* Leave the default **Push events** selected
+* Finish by clicking **Add Webhook**
 
-If you ever need to manually set the webhook on Bitbucket,
-you can point it at ``https://readthedocs.org/bitbucket``.
+Using the generic API integration
+---------------------------------
 
-GitLab
----------
+For repositories that are not hosted with a supported provider, we also offer a
+generic API endpoint for triggering project builds. Similar to webhook
+integrations, this integration has a specific URL, found on the integration
+detail page. This integration can either be added manually or may be created
+automatically if you had previously set up a webhook with your provider
+manually.
```

Comment:
`This integration can either be added manually or may be created
automatically if you had previously set up a webhook with your provider
manually.` Not sure what this is saying. It I previously set it up manually its now automatic?